File: lib/core/models/user_profile.dart
class UserProfile {
  final String? firstName;
  final String? lastName;
  final String? phone;
  final String? avatarUrl;
  final String? bio;
  final DateTime? birthDate;
  final String? gender;
  final String? institution;
  final String? university;
  final int? semester;
  final String? nim;

  const UserProfile({
    this.firstName,
    this.lastName,
    this.phone,
    this.avatarUrl,
    this.bio,
    this.birthDate,
    this.gender,
    this.institution,
    this.university,
    this.semester,
    this.nim,
  });

  factory UserProfile.fromJson(Map<String, dynamic> json) {
    return UserProfile(
      firstName: json['first_name']?.toString(),
      lastName: json['last_name']?.toString(),
      phone: json['phone']?.toString() ?? json['phone_number']?.toString(),
      avatarUrl: json['avatar_url']?.toString(),
      bio: json['bio']?.toString(),
      birthDate: json['birth_date'] != null ? DateTime.tryParse(json['birth_date'].toString()) : null,
      gender: json['gender']?.toString(),
      institution: json['institution']?.toString(),
      university: json['university']?.toString(),
      semester: json['semester'] != null ? int.tryParse(json['semester'].toString()) : null,
      nim: json['nim']?.toString(),
    );
  }

  UserProfile copyWith({
    String? firstName,
    String? lastName,
    String? phone,
    String? avatarUrl,
    String? bio,
    DateTime? birthDate,
    String? gender,
    String? institution,
    String? university,
    int? semester,
    String? nim,
  }) {
    return UserProfile(
      firstName: firstName ?? this.firstName,
      lastName: lastName ?? this.lastName,
      phone: phone ?? this.phone,
      avatarUrl: avatarUrl ?? this.avatarUrl,
      bio: bio ?? this.bio,
      birthDate: birthDate ?? this.birthDate,
      gender: gender ?? this.gender,
      institution: institution ?? this.institution,
      university: university ?? this.university,
      semester: semester ?? this.semester,
      nim: nim ?? this.nim,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'first_name': firstName,
      'last_name': lastName,
      'phone': phone,
      'avatar_url': avatarUrl,
      'bio': bio,
      'birth_date': birthDate?.toIso8601String(),
      'gender': gender,
      'institution': institution,
      'university': university,
      'semester': semester,
      'nim': nim,
    };
  }
}

File: app/Models/UserProfile.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

/**
 * @property int $id
 * @property int $user_id
 * @property string|null $first_name
 * @property string|null $last_name
 * @property string|null $phone
 * @property string|null $avatar_url
 * @property string|null $bio
 * @property string|null $birth_date
 * @property string|null $gender
 * @property string|null $institution
 * @property string|null $university
 * @property int|null $semester
 * @property string|null $nim
 * @property \Illuminate\Support\Carbon|null $created_at
 * @property \Illuminate\Support\Carbon|null $updated_at
 * @property-read \App\Models\User $user
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile newModelQuery()
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile newQuery()
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile query()
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereAvatarUrl($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereBio($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereBirthDate($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereCreatedAt($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereFirstName($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereGender($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereId($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereInstitution($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereLastName($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereNim($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile wherePhone($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereSemester($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereUniversity($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereUpdatedAt($value)
 * @method static \Illuminate\Database\Eloquent\Builder<static>|UserProfile whereUserId($value)
 * @mixin \Eloquent
 */
class UserProfile extends Model
{
    use HasFactory;

    protected static function booted()
    {
        static::saved(function ($profile) {
            $profile->syncUserDisplayName();
        });
    }

    /**
     * Synchronize the parent user's 'name' attribute.
     */
    public function syncUserDisplayName()
    {
        $fullName = trim($this->first_name . ' ' . $this->last_name);
        if ($fullName && $this->user) {
            $this->user->update(['name' => $fullName]);
        }
    }

    protected $fillable = [
        'user_id',
        'first_name',
        'last_name',
        'phone',
        'avatar_url',
        'bio',
        'birth_date',
        'gender',
        'institution',
        'university',
        'semester',
        'nim',
    ];

    /**
     * Get the user that owns the profile.
     */
    public function user()
    {
        return $this->belongsTo(User::class);
    }
}
